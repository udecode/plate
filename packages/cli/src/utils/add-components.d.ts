import type { Config } from '@/src/utils/get-config';
export declare function addComponents(components: string[], config: Config, options: {
    isNewProject?: boolean;
    overwrite?: boolean;
    registryName?: string;
    silent?: boolean;
}): Promise<void>;
//# sourceMappingURL=add-components.d.ts.map