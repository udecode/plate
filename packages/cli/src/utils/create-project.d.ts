import type { initOptionsSchema } from '@/src/commands/init';
import type { z } from 'zod';
export declare function createProject(options: Pick<z.infer<typeof initOptionsSchema>, 'cwd' | 'force' | 'pm' | 'srcDir'>): Promise<{
    projectName: null;
    projectPath: null;
} | {
    projectName: any;
    projectPath: string;
}>;
//# sourceMappingURL=create-project.d.ts.map