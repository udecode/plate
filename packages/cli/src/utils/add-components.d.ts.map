{"version":3,"file":"add-components.d.ts","sourceRoot":"","sources":["add-components.ts"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,MAAM,EAAE,MAAM,wBAAwB,CAAC;AAWrD,wBAAsB,aAAa,CACjC,UAAU,EAAE,MAAM,EAAE,EACpB,MAAM,EAAE,MAAM,EACd,OAAO,EAAE;IACP,YAAY,CAAC,EAAE,OAAO,CAAC;IACvB,SAAS,CAAC,EAAE,OAAO,CAAC;IACpB,YAAY,CAAC,EAAE,MAAM,CAAC;IACtB,MAAM,CAAC,EAAE,OAAO,CAAC;CAClB,iBA0CF"}