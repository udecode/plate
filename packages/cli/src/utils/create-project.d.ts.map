{"version":3,"file":"create-project.d.ts","sourceRoot":"","sources":["create-project.ts"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,iBAAiB,EAAE,MAAM,qBAAqB,CAAC;AAC7D,OAAO,KAAK,EAAE,CAAC,EAAE,MAAM,KAAK,CAAC;AAY7B,wBAAsB,aAAa,CACjC,OAAO,EAAE,IAAI,CACX,CAAC,CAAC,KAAK,CAAC,OAAO,iBAAiB,CAAC,EACjC,KAAK,GAAG,OAAO,GAAG,IAAI,GAAG,QAAQ,CAClC;;;;;;GA4GF"}