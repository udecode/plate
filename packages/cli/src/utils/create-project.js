"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g = Object.create((typeof Iterator === "function" ? Iterator : Object).prototype);
    return g.next = verb(0), g["throw"] = verb(1), g["return"] = verb(2), typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createProject = createProject;
var execa_1 = require("execa");
var fs_extra_1 = __importDefault(require("fs-extra"));
var path_1 = __importDefault(require("path"));
var prompts_1 = __importDefault(require("prompts"));
var get_package_manager_1 = require("@/src/utils/get-package-manager");
var highlighter_1 = require("@/src/utils/highlighter");
var logger_1 = require("@/src/utils/logger");
var spinner_1 = require("@/src/utils/spinner");
function createProject(options) {
    return __awaiter(this, void 0, void 0, function () {
        var proceed, packageManager, _a, name, projectPath, error_1, createSpinner, args, error_2;
        return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    options = __assign({ srcDir: false }, options);
                    if (!!options.force) return [3 /*break*/, 2];
                    return [4 /*yield*/, (0, prompts_1.default)({
                            initial: true,
                            message: "The path ".concat(highlighter_1.highlighter.info(options.cwd), " does not contain a package.json file. Would you like to start a new ").concat(highlighter_1.highlighter.info('Next.js'), " project?"),
                            name: 'proceed',
                            type: 'confirm',
                        })];
                case 1:
                    proceed = (_b.sent()).proceed;
                    if (!proceed) {
                        return [2 /*return*/, {
                                projectName: null,
                                projectPath: null,
                            }];
                    }
                    _b.label = 2;
                case 2:
                    _a = options.pm;
                    if (_a) return [3 /*break*/, 4];
                    return [4 /*yield*/, (0, get_package_manager_1.getPackageManager)(options.cwd, {
                            withFallback: true,
                        })];
                case 3:
                    _a = (_b.sent());
                    _b.label = 4;
                case 4:
                    packageManager = _a;
                    return [4 /*yield*/, (0, prompts_1.default)({
                            format: function (value) { return value.trim(); },
                            initial: 'my-app',
                            message: "What is your project named?",
                            name: 'name',
                            type: 'text',
                            validate: function (value) {
                                return value.length > 128 ? "Name should be less than 128 characters." : true;
                            },
                        })];
                case 5:
                    name = (_b.sent()).name;
                    projectPath = "".concat(options.cwd, "/").concat(name);
                    _b.label = 6;
                case 6:
                    _b.trys.push([6, 8, , 9]);
                    return [4 /*yield*/, fs_extra_1.default.access(options.cwd, fs_extra_1.default.constants.W_OK)];
                case 7:
                    _b.sent();
                    return [3 /*break*/, 9];
                case 8:
                    error_1 = _b.sent();
                    logger_1.logger.break();
                    logger_1.logger.error("The path ".concat(highlighter_1.highlighter.info(options.cwd), " is not writable."));
                    logger_1.logger.error("It is likely you do not have write permissions for this folder or the path ".concat(highlighter_1.highlighter.info(options.cwd), " does not exist."));
                    logger_1.logger.break();
                    process.exit(1);
                    return [3 /*break*/, 9];
                case 9:
                    if (fs_extra_1.default.existsSync(path_1.default.resolve(options.cwd, name, 'package.json'))) {
                        logger_1.logger.break();
                        logger_1.logger.error("A project with the name ".concat(highlighter_1.highlighter.info(name), " already exists."));
                        logger_1.logger.error("Please choose a different name and try again.");
                        logger_1.logger.break();
                        process.exit(1);
                    }
                    createSpinner = (0, spinner_1.spinner)("Creating a new Next.js project. This may take a few minutes.").start();
                    args = [
                        '--tailwind',
                        '--eslint',
                        '--typescript',
                        '--app',
                        options.srcDir ? '--src-dir' : '--no-src-dir',
                        '--no-import-alias',
                        "--use-".concat(packageManager),
                    ];
                    _b.label = 10;
                case 10:
                    _b.trys.push([10, 12, , 13]);
                    return [4 /*yield*/, (0, execa_1.execa)('npx', __spreadArray(['create-next-app@14.2.16', projectPath, '--silent'], args, true), {
                            cwd: options.cwd,
                        })];
                case 11:
                    _b.sent();
                    return [3 /*break*/, 13];
                case 12:
                    error_2 = _b.sent();
                    logger_1.logger.break();
                    logger_1.logger.error("Something went wrong creating a new Next.js project. Please try again.", error_2);
                    process.exit(1);
                    return [3 /*break*/, 13];
                case 13:
                    createSpinner === null || createSpinner === void 0 ? void 0 : createSpinner.succeed('Creating a new Next.js project.');
                    return [2 /*return*/, {
                            projectName: name,
                            projectPath: projectPath,
                        }];
            }
        });
    });
}
