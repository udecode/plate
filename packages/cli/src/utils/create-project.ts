import type { initOptionsSchema } from '@/src/commands/init';
import type { z } from 'zod';

import { execa } from 'execa';
import fs from 'fs-extra';
import path from 'path';
import prompts from 'prompts';

import { getPackageManager } from '@/src/utils/get-package-manager';
import { highlighter } from '@/src/utils/highlighter';
import { logger } from '@/src/utils/logger';
import { spinner } from '@/src/utils/spinner';

export async function createProject(
  options: Pick<
    z.infer<typeof initOptionsSchema>,
    'cwd' | 'force' | 'pm' | 'srcDir'
  >
) {
  options = {
    srcDir: false,
    ...options,
  };

  if (!options.force) {
    const { proceed } = await prompts({
      initial: true,
      message: `The path ${highlighter.info(
        options.cwd
      )} does not contain a package.json file. Would you like to start a new ${highlighter.info(
        'Next.js'
      )} project?`,
      name: 'proceed',
      type: 'confirm',
    });

    if (!proceed) {
      return {
        projectName: null,
        projectPath: null,
      };
    }
  }

  const packageManager =
    options.pm ||
    (await getPackageManager(options.cwd, {
      withFallback: true,
    }));

  const { name } = await prompts({
    format: (value: string) => value.trim(),
    initial: 'my-app',
    message: `What is your project named?`,
    name: 'name',
    type: 'text',
    validate: (value: string) =>
      value.length > 128 ? `Name should be less than 128 characters.` : true,
  });

  const projectPath = `${options.cwd}/${name}`;

  // Check if path is writable.
  try {
    await fs.access(options.cwd, fs.constants.W_OK);
  } catch (error) {
    logger.break();
    logger.error(`The path ${highlighter.info(options.cwd)} is not writable.`);
    logger.error(
      `It is likely you do not have write permissions for this folder or the path ${highlighter.info(
        options.cwd
      )} does not exist.`
    );
    logger.break();
    process.exit(1);
  }

  if (fs.existsSync(path.resolve(options.cwd, name, 'package.json'))) {
    logger.break();
    logger.error(
      `A project with the name ${highlighter.info(name)} already exists.`
    );
    logger.error(`Please choose a different name and try again.`);
    logger.break();
    process.exit(1);
  }

  const createSpinner = spinner(
    `Creating a new Next.js project. This may take a few minutes.`
  ).start();

  // Note: pnpm fails here. Fallback to npx with --use-PACKAGE-MANAGER.
  const args = [
    '--tailwind',
    '--eslint',
    '--typescript',
    '--app',
    options.srcDir ? '--src-dir' : '--no-src-dir',
    '--no-import-alias',
    `--use-${packageManager}`,
  ];

  try {
    await execa(
      'npx',
      ['create-next-app@14.2.16', projectPath, '--silent', ...args],
      {
        cwd: options.cwd,
      }
    );
  } catch (error) {
    logger.break();
    logger.error(
      `Something went wrong creating a new Next.js project. Please try again.`,
      error
    );
    process.exit(1);
  }

  createSpinner?.succeed('Creating a new Next.js project.');

  return {
    projectName: name,
    projectPath,
  };
}
