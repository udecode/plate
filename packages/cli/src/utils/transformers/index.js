"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g = Object.create((typeof Iterator === "function" ? Iterator : Object).prototype);
    return g.next = verb(0), g["throw"] = verb(1), g["return"] = verb(2), typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.transform = transform;
var fs_1 = require("fs");
var os_1 = require("os");
var path_1 = __importDefault(require("path"));
var transform_css_vars_1 = require("@/src/utils/transformers/transform-css-vars");
var transform_import_1 = require("@/src/utils/transformers/transform-import");
var transform_jsx_1 = require("@/src/utils/transformers/transform-jsx");
var transform_rsc_1 = require("@/src/utils/transformers/transform-rsc");
var ts_morph_1 = require("ts-morph");
var transform_tw_prefix_1 = require("./transform-tw-prefix");
var project = new ts_morph_1.Project({
    compilerOptions: {},
});
function createTempSourceFile(filename) {
    return __awaiter(this, void 0, void 0, function () {
        var dir;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, fs_1.promises.mkdtemp(path_1.default.join((0, os_1.tmpdir)(), "shadcn-"))];
                case 1:
                    dir = _a.sent();
                    return [2 /*return*/, path_1.default.join(dir, filename)];
            }
        });
    });
}
function transform(opts_1) {
    return __awaiter(this, arguments, void 0, function (opts, transformers) {
        var tempFile, sourceFile, _i, transformers_1, transformer;
        if (transformers === void 0) { transformers = [
            transform_import_1.transformImport,
            transform_rsc_1.transformRsc,
            transform_css_vars_1.transformCssVars,
            transform_tw_prefix_1.transformTwPrefixes,
        ]; }
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, createTempSourceFile(opts.filename)];
                case 1:
                    tempFile = _a.sent();
                    sourceFile = project.createSourceFile(tempFile, opts.raw, {
                        scriptKind: ts_morph_1.ScriptKind.TSX,
                    });
                    for (_i = 0, transformers_1 = transformers; _i < transformers_1.length; _i++) {
                        transformer = transformers_1[_i];
                        transformer(__assign({ sourceFile: sourceFile }, opts));
                    }
                    if (!opts.transformJsx) return [3 /*break*/, 3];
                    return [4 /*yield*/, (0, transform_jsx_1.transformJsx)(__assign({ sourceFile: sourceFile }, opts))];
                case 2: return [2 /*return*/, _a.sent()];
                case 3: return [2 /*return*/, sourceFile.getText()];
            }
        });
    });
}
