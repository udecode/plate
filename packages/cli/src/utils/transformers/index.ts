import { promises as fs } from 'fs';
import { tmpdir } from 'os';
import path from 'path';
import { Project, ScriptKind, type SourceFile } from 'ts-morph';
import * as z from 'zod';

import { Config } from '../get-config';
import { registryBaseColorSchema } from '../registry/schema';
import { transformCssVars } from './transform-css-vars';
import { transformImport } from './transform-import';
import { transformRsc } from './transform-rsc';

export type TransformOpts = {
  filename: string;
  raw: string;
  config: Config;
  baseColor?: z.infer<typeof registryBaseColorSchema>;
};

export type Transformer = (
  opts: TransformOpts & {
    sourceFile: SourceFile;
  }
) => Promise<SourceFile>;

const transformers: Transformer[] = [
  transformImport,
  transformRsc,
  transformCssVars,
];

const project = new Project({
  compilerOptions: {},
});

async function createTempSourceFile(filename: string) {
  const dir = await fs.mkdtemp(path.join(tmpdir(), 'plate-'));
  return path.join(dir, filename);
}

export async function transform(opts: TransformOpts) {
  const tempFile = await createTempSourceFile(opts.filename);
  const sourceFile = project.createSourceFile(tempFile, opts.raw, {
    scriptKind: ScriptKind.TSX,
  });

  for (const transformer of transformers) {
    transformer({ sourceFile, ...opts });
  }

  return sourceFile.getFullText();
}
