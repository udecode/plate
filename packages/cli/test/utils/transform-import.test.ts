import { expect, test } from "vitest"

import { transform } from "../../src/utils/transformers"

test("transform import", async () => {
  expect(
    await transform({
      filename: "test.ts",
      raw: `import * as React from "react"
import { Foo } from "bar"
    import { Button } from "@/registry/new-york/ui/button"
    import { Label} from "ui/label"
    import { Box } from "@/registry/new-york/box"

    import { cn } from "@/lib/utils"
    `,
      config: {
        tailwind: {
          baseColor: "neutral",
          cssVariables: true,
        },
        aliases: {
          components: "@/components",
          utils: "@/lib/utils",
        },
      },
    })
  ).toMatchSnapshot()

  expect(
    await transform({
      filename: "test.ts",
      raw: `import * as React from "react"
import { Foo } from "bar"
    import { Button } from "@/registry/new-york/ui/button"
    import { Label} from "ui/label"
    import { Box } from "@/registry/new-york/box"

    import { cn, foo, bar } from "@/lib/utils"
    import { bar } from "@/lib/utils/bar"
    `,
      config: {
        aliases: {
          components: "~/src/components",
          utils: "~/lib",
        },
      },
    })
  ).toMatchSnapshot()

  expect(
    await transform({
      filename: "test.ts",
      raw: `import * as React from "react"
import { Foo } from "bar"
    import { Button } from "@/registry/new-york/ui/button"
    import { Label} from "ui/label"
    import { Box } from "@/registry/new-york/box"

    import { cn } from "@/lib/utils"
    import { bar } from "@/lib/utils/bar"
    `,
      config: {
        aliases: {
          components: "~/src/components",
          utils: "~/src/utils",
        },
      },
    })
  ).toMatchSnapshot()
})
