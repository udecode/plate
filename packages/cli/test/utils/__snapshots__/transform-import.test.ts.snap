// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform import 1`] = `
"import * as React from \\"react\\"
import { Foo } from \\"bar\\"
    import { Button } from \\"@/components/ui/button\\"
    import { Label} from \\"ui/label\\"
    import { Box } from \\"@/components/box\\"

    import { cn } from \\"@/lib/utils\\"
    "
`;

exports[`transform import 2`] = `
"import * as React from \\"react\\"
import { Foo } from \\"bar\\"
    import { Button } from \\"~/src/components/ui/button\\"
    import { Label} from \\"ui/label\\"
    import { Box } from \\"~/src/components/box\\"

    import { cn, foo, bar } from \\"~/lib\\"
    import { bar } from \\"@/lib/utils/bar\\"
    "
`;

exports[`transform import 3`] = `
"import * as React from \\"react\\"
import { Foo } from \\"bar\\"
    import { Button } from \\"~/src/components/ui/button\\"
    import { Label} from \\"ui/label\\"
    import { Box } from \\"~/src/components/box\\"

    import { cn } from \\"~/src/utils\\"
    import { bar } from \\"@/lib/utils/bar\\"
    "
`;
